Userland/Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -319,7 +319,7 @@ Vector<MatchingRule> StyleComputer::collect_matching_rules(DOM::Element const& e
             }
         } else {
             for (auto const& rule : rules) {
-                if (filter_namespace_rule(element, rule))
+                if (!rule.contains_pseudo_element && filter_namespace_rule(element, rule))
```

Comment:
I think we'll need to be a bit smarter with `contains_pseudo_element` once we have things like `:has()`, because a selector containing a pseudo-element doesn't necessarily mean its target **is** a pseudo-element. But that will involve renaming that field so we'll catch the problems when that happens. This is fine for now.

Reply:
True, although in fact, `contains_pseudo_element` is really more like `rightmost_selector_must_match_some_kind_of_pseudo_element` even now!